src/components/Hoverable/ActiveHoverable.tsx

```diff
@@ -0,0 +1,143 @@
+import {cloneElement, forwardRef, MutableRefObject, Ref, useCallback, useEffect, useImperativeHandle, useMemo, useRef, useState} from 'react';
+import {DeviceEventEmitter} from 'react-native';
+import CONST from '@src/CONST';
+import HoverableProps from './types';
+
+/**
+ * Assigns a ref to an element, either by setting the current property of the ref object or by calling the ref function
+ * @param ref The ref object or function.
+ * @param element The element to assign the ref to.
+ */
+function assignRef(ref: ((instance: HTMLElement | null) => void) | MutableRefObject<HTMLElement | null>, element: HTMLElement) {
```

Comment:
Is there no built-in type for something like `((instance: HTMLElement | null) => void) | MutableRefObject<HTMLElement | null>`? It's a standard form of a React reference, right?

Reply:
Isn't this just `Ref`?

Reply:
The `Ref` definition contains `RefObject` type which holds `readonly` value. This definition overshadows the `MutableRefObject` which leads to incorrect inference (TS error). Thus we need to combine those.

```
type Ref<T> = RefCallback<T> | RefObject<T> | null;
```

Thanks for pointing that out tho, as the most correct way would be to use `RefCallback` instead of custom function type 👍 

I've changed that ✅